### DISCUSSION TITLE
Initialize command randomly hangs

### DISCUSSION THREAD

**Discussion by zilladotexe** [liquidctl/liquidctl#383] [Q&A]
I'll post this here as I don't really have enough concrete information to file a bug report. I have a script that runs on boot to set pump speed on my Kraken X53. The first line of the script is;

`liquidctl --match kraken initialize`

Now very infrequently (I'm talking once every month or two), this statement will just hang and not complete without printing anything to the console. My only thoughts are that perhaps my script runs too early and the device isn't registered yet but if that were the case, I'm sure liquidctl would tell me it failed to find a matching device.

Any thoughts on how I could potentially debug this or perhaps replicate it to get more information?

Windows 10
liquidctl 1.7.1 x86_64

Thanks all, keep up the great work.

**Comment by jonasmalacofilho** [comment]
I think you're right that the script may be running too soon.

Unfortunately, the device being discoverable (at the USB level) doesn't mean that its firmware is ready to take commands.

**Comment by Maid3n** [comment]
Hello,
I had the same issue, device not detected at startup.
Added a little 3 seconds pause at the beginning of the batch file:
timeout /t 3 /nobreak > NUL
